### PR TITLE
[text analytics] Update ta tests

### LINKED ISSUE #11442: [text analytics] test issues
- [x] switch instances of TextDocumentInput in tests for LD to DetectLanguageInput: https://github.com/Azure/azure-sdk-for-python/blob/feature/text_analytics_v3.0/sdk/textanalytics/azure-ai-textanalytics/tests/test_detect_language.py#L62
- [x] service has fixed, remove skip: https://github.com/Azure/azure-sdk-for-python/blob/feature/text_analytics_v3.0/sdk/textanalytics/azure-ai-textanalytics/tests/test_detect_language.py#L516
- [x] add test for client with null endpoint
- [x] pass in invalid language to appropriate endpoints
- [x] pass in null for docs to all endpoints
- [x] make sure elements returned in same order
- [x] make sure inner error takes precedence over outer error
- [x] remove duplicate bad_model_version tests

### PR DESCRIPTION
**Pull request by iscai-msft**
fixes #11442 
definitely recommend looking commit by commit since they're independent of each other

### CODE REVIEW

**Comment by maririos** [comment]
I'm curious... was this driven by looking at the tests other languages have?

File: sdk/textanalytics/azure-ai-textanalytics/azure/ai/textanalytics/_request_handlers.py [hunk 1]
```diff
@@ -32,6 +32,9 @@ def _validate_batch_input(documents, hint, whole_batch_hint):
         if not all(isinstance(x, (dict, TextDocumentInput, DetectLanguageInput)) for x in documents):
             raise TypeError("Mixing string and dictionary/object document input unsupported.")
 
+    if not isinstance(documents, list):
+        raise TypeError("`documents` parameter must be a list, not a {}".format(type(documents)))
```
Comment (kristapratico) [inline sdk/textanalytics/azure-ai-textanalytics/azure/ai/textanalytics/_request_handlers.py:36]: we shouldn't do this type of validation - a user should be able to pass any iterable

File: sdk/textanalytics/azure-ai-textanalytics/tests/test_analyze_sentiment.py [hunk 2]
```diff
@@ -121,15 +137,6 @@ def test_bad_credentials(self, client):
                 ["This is written in English."]
             )
 
-    @GlobalTextAnalyticsAccountPreparer()
-    @TextAnalyticsClientPreparer()
-    def test_bad_model_version(self, client):
```
Comment (kristapratico) [inline sdk/textanalytics/azure-ai-textanalytics/tests/test_analyze_sentiment.py:126]: are we deleting because the test is duplicated?

**Comment by iscai-msft** [comment]
> I'm curious... was this driven by looking at the tests other languages have?

yep, for the most part!

File: sdk/textanalytics/azure-ai-textanalytics/tests/test_analyze_sentiment.py (hunk 2 above)
Comment (iscai-msft) [reply sdk/textanalytics/azure-ai-textanalytics/tests/test_analyze_sentiment.py:126]: yep, we had test_bad_model_version_error and test_bad_model_version. Deleting test_bad_model_version since test_bad_model_version_error goes more into the returned error and is more detailed

File: sdk/textanalytics/azure-ai-textanalytics/azure/ai/textanalytics/_request_handlers.py (hunk 1 above)
Comment (iscai-msft) [reply sdk/textanalytics/azure-ai-textanalytics/azure/ai/textanalytics/_request_handlers.py:36]: Ok, I see what you're saying. I added this because I accidentally passed a dict instead of a dict in a list ({"id": 1, "text": "hello", "language": "es"}), and it went through, but passed in three documents to the generated code (one for each entry in the dict).

I'll at least add the check to make sure documents isn't a dict (since that's an iterable). Do you think there's anything else I should check?
